src/vm.rs

```diff
@@ -144,12 +144,18 @@ impl<'r, 't> Nfa<'r, 't> {
                     break
                 }
 
+                // If the expression starts with a '^' we can terminate as soon
+                // as the last thread dies.
+                if self.ic != 0 && prefix_anchor {
+                    break;
+                }
+                
                 // If there are no threads to try, then we'll have to start
                 // over at the beginning of the regex.
                 // BUT, if there's a literal prefix for the program, try to
                 // jump ahead quickly. If it can't be found, then we can bail
                 // out early.
-                if self.prog.prefix.len() > 0 && clist.size == 0 {
```

Comment:
I'm kind of concerned that this doesn't cause a test to fail... Perhaps I need to come up with one. :-)